Pinned item visibility and temporal item visibility fixies

luriusTM opened the pull request:
Removed: 
- Temporal item reorder possibility. (this is not implemented and cause more harm than gain)

Fix: 
- The sync for monitor in case of pinned / unpinned items were broadcasted to FR instead of target update (on frontend side)
- On monitor change there were no update, it is now given.
- Change only synchronized up, when there is anything to communicate to.
- Initial synchronization on app start for WEG was not based on monitor, this has fulfilled.

Feat: 
- Possibility to hide pinned items on not primary monitors (overall & by monitor settings).
- On reduced monitors, where only temporal items, show items without separators
- On reduced monitors, do not let the pin & unpin, because these WEG bar does not have information about this.
- On reduced monitors do not let to store pinned items on reorder
- Possibility to prohibit the pin & unpin option from a temporal | pinned item.

Dependency: 
https://github.com/Seelen-Inc/slu-lib/pull/8

Main monitor: 
![image](https://github.com/user-attachments/assets/39c73058-a07c-4a74-ab95-1ffdcd880a67)
![image](https://github.com/user-attachments/assets/a633ae56-15c7-4431-adce-fe5b7570a708)

On non primary monitor with settings for this: 
![image](https://github.com/user-attachments/assets/a9907ad2-0fc0-4d50-bb14-51159d3e2f6f)
![image](https://github.com/user-attachments/assets/9879659b-a968-40a8-9ff9-72479662f8d6)
![image](https://github.com/user-attachments/assets/de829849-3c2d-4fbb-9431-5bdc2775df20)

eythaann commented on src/apps/seelenweg/modules/item/infra/DraggableItem.tsx:
```diff
@@ -18,7 +20,7 @@ export function DraggableItem({ children, item, className }: Props) {
       ref={ref}
       value={item}
-      drag
+      drag={isPinnedApp(item)}
```
Drag is ok for temporal items, let it always enabled, this should work as currently is, in your feature just allow move items but not call the store function.

luriusTM replied:
Here the why is easy. Because it is not implemented. When you reorder something any any operation happens on any bar it is immediatelly orders everything back. This can cause the issue with the AHK keys (when you press Win + 1), that there is different app should react to this key. 

So: 
- The reorder is not functionating, because it randomly can disappear
- The inconsistance in rust and JS is not tolerable. 

I'm gladly implement the reorder functionality and enables it back, it just needs a command and a better handling of communicated items in rust. This is simple and easy, but should not be part of this PR...

As maybe this should not also but... the two side has a representation difference. While on rust side you have left & right, center, on JS side you have this 3 group with separator of Separators... if I remove this separator then there is a problem... I can hijact that, but would be much better to implement it via normal reoder function. There needs some implementation here or there for sure. This is the easiest and more fit the current state.

luriusTM replied:
Or I can remove this handler to just avoid the conflict and could implement it later on... every other has already been implemented... just this seemed more like the reality... please make a decision and will make it as you wish.

eythaann replied:
Allow move, we can search another alternative for the Win + Number feature.

luriusTM replied:
Not the win + number is the biggest harm... this is the problem: 
- The reorder is not functionating, because it randomly can disappear
- The inconsistance in rust and JS is not tolerable.
 

Maybe a video...

https://github.com/user-attachments/assets/b8df91a3-1567-4c67-95cf-b32d34d92b24

luriusTM replied:
It can be moved again.

eythaann commented on src/apps/seelenweg/modules/bar/index.tsx:
```diff
@@ -157,43 +163,49 @@ export function SeelenWeg() {
         className={cx('taskbar', settings.position.toLowerCase(), {
           horizontal: isHorizontal,
           vertical: !isHorizontal,
+          'temporal-only': isTemporalOnlyWegBar,
```
Or use it on the default theme, also I don't see any case of no items, the dock have margins and these handle mouse events so is a ghost collision box, check this.

luriusTM replied:
This is information which can be outsourced to CSS. I would use it for styling, so this information is useful. 

I can follow this pattern for now or please describe what would be the default theme expectation or differences in this case... (as this is a simple state, which does not have an addition in normal way, I think just should be used to modify appearances. As much this is a simple serious theme... I think nothing is needed)
![image](https://github.com/user-attachments/assets/354496cf-c338-44a2-b090-f897e3ca92ae)

luriusTM commented:
Added an empty state message:
![image](https://github.com/user-attachments/assets/16bea582-75e6-45d3-9ee0-292abca4fb39)

Added a reduced context menu for this type of weg bar: 
![image](https://github.com/user-attachments/assets/dac533e2-cc8b-4e4c-b87d-43238132f256)

Normal state: 
![image](https://github.com/user-attachments/assets/efce01ca-626a-44fc-aaf2-1324ca139ff4)
